src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -620,8 +629,11 @@ protected function buildDeployment(Device $deviceForFunctions, Webhook $queueFor
             $durationEnd = \microtime(true);
 
             $buildSizeLimit = (int)System::getEnv('_APP_FUNCTIONS_BUILD_SIZE_LIMIT', '2000000000');
+            if (isset($plan['buildSize'])) {
+                $buildSizeLimit = $plan['buildSize'] * 1000 * 1000 * 1000;
```

Comment:
Let's keep it as MB too,

```suggestion
                $buildSizeLimit = $plan['buildSize'] * 1000 * 1000;
```